shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,171 @@
+                ManagedProcessor.COMPONENT_NAME,
+                CamelIntegration.class));
+
+        return eventSources;
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        String managedProcessorName = managedProcessor.getMetadata().getName();
+        String managedProcessorNamespace = managedProcessor.getMetadata().getNamespace();
+
+        LOGGER.info("Reconciling ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        ManagedProcessorStatus processorStatus = managedProcessor.getStatus();
+
+        if (!processorStatus.isReady() && isTimedOut(processorStatus)) {
+            // notifyManagerOfFailure
+            processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        String integrationName = String.format("integration-%s", managedProcessorName);
+        CamelIntegration camelIntegration = managedProcessorService.fetchOrCreateCamelIntegration(managedProcessor, integrationName);
+
+        if (!camelIntegration.isReady()) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is not ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            if (!processorStatus.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                processorStatus.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            return UpdateControl.updateStatus(managedProcessor).rescheduleAfter(executorPollIntervalMilliseconds);
+        } else if (!processorStatus.isConditionTypeTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE)) {
+            LOGGER.info("CamelIntegration for the ManagedProcessor '{}' in namespace '{}' is ready",
+                    managedProcessorName,
+                    managedProcessorNamespace);
+            processorStatus.markConditionTrue(ManagedProcessorStatus.CAMEL_INTEGRATION_AVAILABLE);
+        }
+
+        LOGGER.info("Managed Processor: '{}' in namespace '{}' is ready",
+                managedProcessorName,
+                managedProcessorNamespace);
+
+        // Only issue a Status Update once.
+        // This is a work-around for non-deterministic Unit Tests.
+        // See https://issues.redhat.com/browse/MGDOBR-1002
+        if (!managedProcessor.getStatus().isReady()) {
+            metricsService.onOperationComplete(managedProcessor, MetricsOperation.CONTROLLER_RESOURCE_PROVISION);
+            processorStatus.markConditionTrue(ConditionTypeConstants.READY);
+            // Notify Manager is Ready
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        return UpdateControl.noUpdate();
+    }
+
+    private boolean isTimedOut(ManagedProcessorStatus status) {
```

Comment:
FYI, we need to discuss https://issues.redhat.com/browse/MGDOBR-1308 to decide if we want to keep timeout logic in the Data Plane or whether it makes sense to fully delegate that to the Control Plane. If we do decide to keep  sometime logic in the data plane, then I think we can push this method onto a super-class that all our `Controllers` extend. We'll also need to decide how we communicate a timeout via `Conditions`. 

I'm personally happy for you to exclude the timeout logic from this PR until we've had that discussion, but up-to-you.